aspnetcore/security/authorization/secure-data/samples/final15/Authorization/ContactRoleAuthorizationHandler.cs

```diff
@@ -16,7 +16,7 @@ public class ContactRoleAuthorizationHandler
                 return Task.FromResult(0);
             }
 
-            if (context.User.IsInRole(requirement.Name))
+            if (context.User.IsInRole(Constants.OperationRolePrefix + requirement.Name))
```

Comment:
So I'd end up with nested handlers. You would have a policy that checks if the user can access Contacts at all. Then for the edit you would use declarative to check the current user is in Contacts, and then can Edit, then finally, inside edit you'd to resource based authorization imperatively, pass the contact in, and then have the admin handler apply to both that requirement, and a requirement that checks the current user ID against the resource owner. 

So you end up with an admin handler which copes with multiple requirements/operations (because operations are just requirements *grin*)